src/smt/theory_str_noodler/nielsen_decision_procedure.cpp

```diff
@@ -304,14 +297,9 @@ namespace smt::noodler {
         return false;
     }
 
-    /**
-     * @brief Create a counter system from the Nielsen graph.
-     * 
-     * @param graph Graph to be converted to the counter system.
-     * @return CounterSystem
-     */
-    CounterSystem NielsenDecisionProcedure::create_counter_system(const NielsenGraph& graph) const {
```

Comment:
I think a better practice is to return `bool` and as an output parameter set the counter system.

Reply:
I changed it (also `get_length_path`), check if it is ok.

Reply:
Great! I have just updated the corresponding comments.